### PR TITLE
FLPATH-243: Code clean in workflow aspect

### PR DESCRIPTION
**Pull request by RichardW98**
1. Code clean in workflow aspect
2. rename WorkStatus to ParodosWorkStatus as it's reserved in workflow-engine

### CODE REVIEW

File: workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowTaskExecutionAspect.java [hunk 1]
```diff
@@ -99,90 +98,95 @@ public WorkReport executeAroundAdviceTask(ProceedingJoinPoint proceedingJoinPoin
...
 
 		try {
 			report = (WorkReport) proceedingJoinPoint.proceed();
+			if (report == null || report.getStatus() == null)
+				throw new NullPointerException("task execution not returns status: " + workFlowTaskName);
 		}
 		catch (Throwable e) {
-			log.error("Workflow task execution {} has failed!", workFlowTaskName);
+			log.error("Workflow task execution {} has failed! error message: {}", workFlowTaskName, e.getMessage());
+			report = new DefaultWorkReport(WorkStatus.FAILED, workContext);
```
[the remark this answers is not in the document]
Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowTaskExecutionAspect.java:140]: I think we should persist even it's failed with exception

File: workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java [hunk 2]
```diff
@@ -125,10 +125,17 @@ public WorkFlowExecution getWorkFlowById(UUID workFlowExecutionId) {
...
-		return workFlowRepository.save(WorkFlowExecution.builder().workFlowDefinitionId(workFlowDefinitionId)
-				.projectId(projectId).status(workFlowStatus).startDate(new Date())
-				.masterWorkFlowExecution(masterWorkFlowExecution).build());
+			WorkFlowStatus workFlowStatus, WorkFlowExecution masterWorkFlowExecution, String arguments) {
+		try {
+			return workFlowRepository.save(WorkFlowExecution.builder().workFlowDefinitionId(workFlowDefinitionId)
+					.projectId(projectId).status(workFlowStatus).startDate(new Date()).arguments(arguments)
+					.masterWorkFlowExecution(masterWorkFlowExecution).build());
+		}
+		catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java:134]: it's for all DB operation exception, no specific one.

File: workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java [hunk 3]
```diff
@@ -125,10 +125,17 @@ public WorkFlowExecution getWorkFlowById(UUID workFlowExecutionId) {
...
+			log.error("failing persist workflow execution for: {} in master workflow execution: {}",
```
[the remark this answers is not in the document]
Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java:135]: yes I will print the exception message

File: workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (pkliczewski) [reply workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java:134]: Is there any exception hierarchy we could use? This seems to generic.

File: workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowTaskExecutionAspect.java [hunk 4]
```diff
@@ -99,90 +98,95 @@ public WorkReport executeAroundAdviceTask(ProceedingJoinPoint proceedingJoinPoin
...
 
-		// skip the task if it's already successful
 		UUID masterWorkFlowExecutionId = UUID.fromString(WorkContextDelegate
 				.read(workContext, WorkContextDelegate.ProcessType.WORKFLOW_EXECUTION, WorkContextDelegate.Resource.ID)
 				.toString());
 
-		WorkFlowExecution masterWorkFlowExecution = workFlowRepository.findById(masterWorkFlowExecutionId).get();
-		// get the workflow if it's executed again from continuation
+		WorkFlowExecution masterWorkFlowExecution = workFlowRepository.findById(masterWorkFlowExecutionId)
+				.orElseThrow(() -> new RuntimeException("masterWorkFlow not found for task: " + workFlowTaskName));
```
[the remark this answers is not in the document]
Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/workflow/execution/aspect/WorkFlowTaskExecutionAspect.java]: fixed

File: workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java]: fixed

File: workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImpl.java]: fixed.   jpa exception hierarchy root "DataAccessException" is added

**Review by pkliczewski — COMMENTED** [review]
/lgtm

**Pull request merged**